### DISCUSSION TITLE
Aplicando Orientação a Objetos em Projetos Reais com C# 11 e .NET 7 - Value Objects

### DISCUSSION THREAD

**Discussion by weslleyrufino** [balta-io/forum#409] [Tira Dúvidas] [answered]
Falaí Balta, tudo bem?

Tenho uma dúvida, parece ser insignificante, mas quero perguntar mesmo assim. 
Nessa aula do título você cria uma classe ValueObject com o objetivo que se eu quiser saber quais são os objetos do tipo ValueObject, eu consigo saber através dessa referência ao herdar a mesma.
É simplesmente esse objetivo? Se as outras classes (Campaign.cs, Url.cs) já estão na pasta ValueObjects isso tb já não teria o mesmo sentido?
A idéia é manter a classe ValueObject.cs vazia mesmo?

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @weslleyrufino, como vai?
Não é insignificante não, é válida sim..

Os VOs tem como objetivo concentrar comportamentos também, não é simplesmente propriedades. Um VO de E-mail por exemplo, contém a validação do mesmo e por aí vai.

Estes VOs compõe outras entidades, eles são super importantes.

Em relação ao VO base, é normal ter uma classe ou interface base para os objetos mais importantes, mesmo que ela não tenha implementação inicialmente, ela serve para agrupar estes objetos.

**Reply by weslleyrufino** [reply]
Estou bem e você?

Entendi. Faz sentido, pois se eu quiser mover VOs filhos para outro lugar eles ainda vão estar herdando a VO pai e mantendo a relação.
Obrigado!
